GetStream/stream-chat-android
# [2959] Update UI Components documentation structure to match Compose

Closes #2959 

### 🎯 Goal

We need to restructure our UI components documentation to match Compose.

### 🛠 Implementation details

- Created the "General Customization" category
- Created the "Channel Components" category
- Created the "Message Components" category
- Created the "Utility Components" category

### 🎨 UI Changes

| Before | After |
| --- | --- |
| <img width="587" alt="Screenshot 2022-03-09 at 12 14 04" src="https://user-images.githubusercontent.com/9600921/157410324-be2eb301-3182-495a-96c3-e66c9d6208d0.png"> | <img width="592" alt="Screenshot 2022-03-09 at 12 08 44" src="https://user-images.githubusercontent.com/9600921/157410034-b5358d6f-516d-4b3a-89ea-9cd597803725.png"> |
| ![screencapture-localhost-3000-chat-docs-sdk-android-ui-overview-2022-03-09-12_15_41](https://user-images.githubusercontent.com/9600921/157410634-b6340ccc-892b-42cc-b297-f50e6290bb2a.png) | ![screencapture-localhost-3000-chat-docs-sdk-android-ui-overview-2022-03-09-12_11_05](https://user-images.githubusercontent.com/9600921/157409930-da5cd47a-2a35-4de9-a372-8e5dbcd267eb.png) |

### 🧪 Testing

1. Run the `npx stream-chat-docusaurus -i -s` command
2. Observe the updated documentation

### ☑️Contributor Checklist

#### General

#### Code & documentation

### ☑️Reviewer Checklist
